[css-text] Rework some tests to avoid anti aliasing issues

frivoal opened the pull request:
As described in https://bugzilla.mozilla.org/show_bug.cgi?id=1498698#c1
The tests introduced in https://github.com/web-platform-tests/wpt/pull/11373
fail due to anti aliasing issues that let some red show through at the edges
even when things are correctly aligned and the red should be hidden.

Rework these tests to avoid the problem.

foolip commented:
@frivoal looks like there are conflicts. I tried to resolve them but they're not trivial enough. Can you resolve and I'll do light review?

frivoal committed:
[css-text-3] Rework text-indent tests

These tests used to have anti aliasing artefacts. A previous commit
(cf1491254cca) solved that, at the expense of making the tests
impractical to review manually, and no longer self descriptive.

This change fixes that, while keeping the tests robust to anti aliasing
artefacts.
